lib/tessel/wifi.js

```diff
@@ -163,7 +163,18 @@ Tessel.prototype.setWiFiState = function(enable) {
   return new Promise((resolve, reject) => {
     return this.simpleExec(commands.turnOnWifi(enable))
       .then(() => this.simpleExec(commands.commitWirelessCredentials()))
-      .then(() => this.simpleExec(commands.reconnectWifi()))
+      .then(() => {
+        return this.simpleExec(commands.reconnectWifi())
+          .then((result) => {
+            // check if the actual wifi radio (wireless.radio0) is disabled
+            if (result.includes("'radio0' is disabled")) {
+              return this.simpleExec(commands.turnRadioOn())
+                .then(() => this.simpleExec(commands.commitWirelessCredentials()))
+                .then(() => this.simpleExec(commands.reconnectWifi()))
+                .catch(reject);
+            }
```

Comment:
I believe, and after experimenting, once the execution of a `thenable` function finishes, the Promise will continue to the next chained method. Returning in a `thenable` function will pass that value to the next `thenable`.